[13.0][FIX] account_invoice_date_due: restore multi-record handling in write()

joao-p-marques opened the pull request:
415bb0b4c2eea0435fbd6431dad8b239216c90dd introduced a bug disallowing to `write()` in a multi-record set.
The bug is reproduced when doing mass edits.
This restores that possibility.

@Tecnativa
TT31218

ping @pedrobaeza @victoralmau

pedrobaeza commented on account_invoice_date_due/models/account_move.py:
```diff
@@ -21,10 +21,14 @@ def write(self, vals):
-            )
-            payment_term_lines.write({"date_maturity": vals["invoice_date_due"]})
+        if "invoice_date_due" in vals:
+            for record in self:
+                if record.state == "posted":
```
```suggestion
            for record in self.filtered(lambda x: x.state == "posted"):
``

joao-p-marques replied:
Thanks! Change made

joao-p-marques committed:
[FIX] account_invoice_date_due: restore multi-record handling in write()

415bb0b4c2eea0435fbd6431dad8b239216c90dd introduced a bug disallowing to write() in a multi-record set.
The bug is reproduced when doing mass edits.
This restores that possibility.

TT31218

pedrobaeza commented:
Let's fast-track it for avoiding the singleton error:

/ocabot merge patch
